examples/timestamp/clock_set.c

```diff
@@ -0,0 +1,184 @@
+/* clock_set.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfTPM.
+ *
+ * wolfTPM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfTPM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* This example shows how to increment the TPM2 clock */
+
+#include <wolftpm/tpm2_wrap.h>
+
+#ifndef WOLFTPM2_NO_WRAPPER
+
+#include <examples/tpm_io.h>
+#include <examples/tpm_test.h>
+#include "clock_set.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/******************************************************************************/
+/* --- BEGIN TPM Clock Set Example -- */
+/******************************************************************************/
+
+static void usage(void)
+{
+    printf("Expected usage:\n");
+    printf("./examples/clock/clockSet [time]\n");
+    printf("* time is value in miliseconds\n");
+    printf("\tThe TPM clock can be set only forward.\n");
+    printf("\tThe TPM clock shows the total time the TPM was ever powered.\n");
+    printf("\tThe TPM clock is different and always higher than the\n");
+    printf("\tcurrent uptime. The TPM uptime can not be tampered with.\n");
+    printf("Demo usage without parameters:\n");
+    printf("* increments the TPM clock with 50 seconds (50,000 ms)\n");
+}
+
+int TPM2_ClockSet_Test(void* userCtx, int argc, char *argv[])
+{
+    int rc = 0;
+    WOLFTPM2_DEV dev;
+
+    union {
+        ClockSet_In clockSet;
+        byte maxInput[MAX_COMMAND_SIZE];
+    } cmdIn;
+    union {
+        ReadClock_Out readClock;
+        byte maxOutput[MAX_RESPONSE_SIZE];
+    } cmdOut;
+
+    TPMS_AUTH_COMMAND session[MAX_SESSION_NUM];
+
+    UINT64 oldClock, newClock;
+
+    if (argc == 2) {
+        newClock = atoi(argv[1]);
+    }
+    else if (argc == 1) {
+        newClock = 0;
+    }
+    else {
+        printf("Incorrect arguments\n");
```

Comment:
The only way to see the usage is to enter more than 1 args. Could you add a way to parse `?` to show the usage?

Reply:
Ah, yes. I forgot to add that option. We have something similar in the other examples. I think it was -h @dgarske ? Will check and add.